Feature/utc 285 patternlab card component

UTCbridget opened the pull request:
This is the initial start of the card component for Pattern Lab. I'm already up to 45, so we'll need to discuss how we'd like to present. This may be okay for the Steering Committee as is, but upon final deployment, we probably need to reorganize it. I'm open to suggestions.

Here's what it looks like so far: 

White BG:

![utccard-pl-component-white](https://user-images.githubusercontent.com/82905787/128936949-bf96b184-aaf4-4618-8173-9e9ada6b40db.gif)

Lt. Blue BG:

![utccard-pl-component-ltblue](https://user-images.githubusercontent.com/82905787/128936989-e9c646b6-3bf6-4c8e-8a13-0de446440a52.gif)

No BG:

![utccard-pl-component-none](https://user-images.githubusercontent.com/82905787/128937053-4945766a-8f86-499f-add9-53ed9fe19ca3.gif)

NOTE: We probably should remove/disable the white-outline option for the button on cards. It's really not working on the any of the backgrounds.

Also note: This doesn't include wide cards, cards with font awesome, vertical cards, card link without button, etc. Those need to be done, but I think this is sufficient to show the Steering Committee.

UTCbridget committed:
Create utccards.twig

Create PL component for cards

UTCbridget committed:
Create index.js

Add PL card component

UTCbridget committed:
Create utccards.yml

Add PL card component

UTCbridget committed:
Create utccards~ltbluebackground.yml

Add PL card component

UTCbridget committed:
Create utccards~nobackground.yml

Add PL card component

UTCbridget committed:
Create utccard.test.js

Add PL card component

UTCbridget committed:
Create _utccard.twig

Add PL card component

brittrich commented on source/default/_patterns/02-molecules/utccard/_utccard.twig:
```diff
@@ -0,0 +1,95 @@
+
+	{% if utccard.card_link %}
+     </a>
+  {% endif %}
```
Are there future plans to take this and make it tailwind vs. the css classes?  Maybe in further redesigns? I do know I've found some things to be trickier in tailwind (e.g. the way this one seamlessly modifies classes for alignment, width, etc. based on user input without having to use a bunch of if statements)...and some things to be easier. Just curious. This one was an OG from pre-tailwind days.

UTCbridget replied:
I think your suggestions are great! Thank you! I just wasn't sure if I needed to include all options bc if developers need the code from "Expand" button, they might have to recode it themselves if they want alternative solutions.

I used tailwinds where I could, but left existing classes to keep the integrity of the existing layout. I can look at converting them, though. Thoughts @bmartinez287?

UTCbridget replied:
I'll take a look at the button taxonomy and see what I can do there.

bmartinez287 replied:
So, it be great if could simplify and refactor some of the code. I think a feature as complex and as useful as cards could use it. It be great exploring which items could be updated to tailwindcss and which ones might be too hard to rewrite/update.

In regards to the pattern-lab demo, we can probably break a card into multiple card types and then put them together based on certain if statements. Like sideimagecard.twig, topimagecard.twig
That should make cards easier to visualise and maybe reuse in code. We probably need to do more reasearch into standard naming conventions and way others are handling such complex items.

UTCbridget replied:
I'm also wondering... if we do the buttons in the atoms, maybe we don't need all the button examples because developers would know (and we could add instructions) to replace any button code with their new button code. That would eliminate a good bit of the cards and keep the page focused on actual card variance.

bmartinez287 replied:
Yeah, i don't think we need the text variants or buttons or other smaller items. We just need the large layout variations.

brittrich replied:
@UTCbridget let me know if you need anything from me if you decide to refactor the cards with some more tailwind. Happy to help, but also happy to let you take the lead on it if it's a useful way for you to get to know that element and what could be done to improve it! Just lmk either way.

UTCbridget replied:
@brittrich, could you do the buttons? I'm going to redo the cards without all the button variations. I'll look at the refractoring and see what can be done there.

brittrich replied:
Sounds good to me! I'll work on buttons next.

UTCbridget replied:
I have reworked the PL Cards with different variations combined per our conversation above and separated them out into Horizontal and Vertical cards. Instructions have been provided, indicating that card features can be mixed and matched. I have also refactored the cards to TailwindCSS. 

**Vertical Cards:** 

![utccard-pl-component-vt](https://user-images.githubusercontent.com/82905787/129774653-554be88f-e959-4399-a7ca-675b0ff7bf45.gif)

**Horizontal Cards:**

![utccard-pl-component-hz](https://user-images.githubusercontent.com/82905787/129774707-e997416c-cc41-4522-b6ee-a6d5638a862f.gif)

Branded Welcome: 

![Screen Shot 2021-08-17 at 1 10 32 PM](https://user-images.githubusercontent.com/82905787/129774741-c2e2f068-3682-4db8-b4d8-9917674db6ff.png)

I look forward to your thoughts!

brittrich replied:
I think this looks great, Bridget! 

Last thought...and it may not be a useful one....but what if the Titles of the Cards reflected the key feature that's being demoed? For example, Left Align / Center Align / Right Align...or....75% Width Card / 50% Width Card / etc.

I've been looking at this design system some for the way they do the buttons. The way they handle cards is also interesting, looks like they have alot of variations as well: https://boltdesignsystem.com/pattern-lab/?p=viewall-components-card 

Can I ask, PL newb that I am, what's the purpose of the code snip/example in the context of our design system? Is it so that folks can use/recreate the component in another system, perhaps? Or, so we can see how everything connects in the code at a glance?

I'm learning a lot from this PR and Bernardo's PR for the Hero as I work on getting the buttons added! Hopefully will have something to show there by the end of this week that can maybe be integrated up to the Cards.

bmartinez287 replied:
It has a few.

1. Provide consistent design for Drupal and WordPress as they would tie into the same components.
2. Provide a kitchen sink of features for vendors such as https://gomocs.com/  https://securelb.imodules.com/s/1341/utc/form/19/form.aspx?sid=1341&gid=3&pgid=4310&cid=6419
3. Allow IT and other tech-based units on campus to have a visual representation of branding and a living example they could try to use or imitate. 


What Figma does for graphic designers on campus, pattern lab will do for developers, vendors, and internal resources.
Figma is supposed to give the flexibility to make quick visual changes based on a designer's input.

Patternlab might even be useful for some people on campus as they might be able to see some sample page layouts for homepages and choose if they want to recreate those or do their own.

brittrich replied:
Nice! Makes sense! Thanks, Bernardo.

brittrich reviewed (approved):
@UTCbridget, this is so awesome!

Related to this: "NOTE: We probably should remove/disable the white-outline option for the button on cards. It's really not working on the any of the backgrounds."

You are right! It'd be nice to disable that. The button options pull from the button taxonomy that is also used in other places (eg. button group)...is there is a way to modify what options from the taxonomy are available to this specific element?

As to the organization of these in PL, I wonder if it'd be less content but same effect to have something like:

**Background and Alignment Options**
[White BG Left w/ Blue button]    [White BG Center w/ Blue button]    [White BG Left w/ Blue button]
[Blue BG Left w/ Blue button]    [Blue BG Center w/ Blue button]    [Blue BG Left w/ Blue button]
[No BG Left w/ Blue button]    [No BG Center w/ Blue button]    [No BG Left w/ Blue button]

**Button Options**
[White BG Left w/ Blue button]   [White BG Left w/ Yellow button]  [White BG Left w/ Green button] 
[White BG Left w/ Light Blue button] [etc....]

Just theorizing if you've seen the way one color button aligns...is that enough as far as alignment? It might make sense to have them paired with each bg option still, though. It might be easier to bring in the other options as well if this were the approach. For example, a horizontal card example might not need to be displayed in every single color and with every single button if those have already been shown. The alignments would still be relevant, though. You all know the committee better than I do, though, so it may absolutely be easiest to have all the options represented in a unique instance for each one...so disregard all of this if that's the case!

Last additional thought is that I like have the data inside each card about what settings it has, that's useful. Curious if there is some use case for having various amounts of text in the cards, though, to show how they respond in various use cases.

UTCbridget committed:
Update welcome.twig

Brand the PL welcome for UTC.

UTCbridget committed:
Create utc-horizontal-logo.png

Add utc png logo to PL.

UTCbridget committed:
Remove original utccards files from PL

Remove original utccards files from PL (We're separating these out into Horizontal and Vertical cards.

UTCbridget committed:
Add UTC Vertical card files to PL

Add UTC Vertical card files to PL

UTCbridget committed:
Add UTC Horizontal card files to PL

Add UTC Horizontal card files to PL

UTCbridget committed:
Update _utccard.twig

Update _utccard.twig with TailwindCSS

UTCbridget committed:
Update tailwind.config.js

Add Tailwind configuration to match existing non-tailwind configuration. Also, FontAwesome css not being picked up in PL (including the CDN all.css file. Had manually add for PL). Add custom FontAwesome sizing to simulate existing configuration.

UTCbridget committed:
Update colors.tailwind.json

Added the link color to custom UTC Tailwind colors.

UTCbridget committed:
Remove old files

Remove old PL card files

UTCbridget commented:
Good idea RE: Card Titles. I will make those changes and PR them today.

UTCbridget committed:
Update card titles and adjust card copy

Update card titles and adjust card copy

UTCbridget committed:
Update _utccard.twig

Removed "Note: The image is currently hardcoded." BC it's not any more.
